src/Raven.Server/Documents/ShardedHandlers/ShardedSmugglerHandler.cs

```diff
@@ -0,0 +1,150 @@
+
+                    result.AddError($"Error occurred during export. Exception: {e}");
+                    await WriteResultAsync(context, result, ResponseBodyStream());
+
+                    HttpContext.Abort();
+                }
+            }
+        }
+
+        public async Task<IOperationResult> ExportShardedDatabaseInternalAsync(
+            DatabaseSmugglerOptionsServerSide options,
+            BlittableJsonReaderObject blittableJson,
+            long startDocumentEtag,
+            long startRaftIndex,
+            Action<IOperationProgress> onProgress,
+            JsonOperationContext jsonOperationContext,
+            OperationCancelToken token)
+        {
+            var tcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            var oldOperateOnType = options.OperateOnTypes;
+
+            var operateOnTypes = options.OperateOnTypes &= ~(DatabaseItemType.DatabaseRecord |
+                                                             DatabaseItemType.Subscriptions |
+                                                             DatabaseItemType.Identities |
+                                                             DatabaseItemType.Indexes |
+                                                             DatabaseItemType.ReplicationHubCertificates);
+
+            blittableJson = CreateNewOptionBlittableJsonReaderObject(blittableJson, jsonOperationContext, nameof(options.OperateOnTypes), operateOnTypes);
+
+            using (var outputStream = GetOutputStream(ResponseBodyStream(), options))
+            using (var exportOutputStream = new GZipStream(outputStream, CompressionMode.Compress))
+            {
+                await exportOutputStream.WriteAsync(Encoding.UTF8.GetBytes($"{{  \"BuildVersion\" : {ServerVersion.Build}"));
+                for (int i = 0; i < ShardedContext.ShardCount; i++)
+                {
+                    if (i == ShardedContext.ShardCount - 1) // Last shard need to bring all server wide information
+                    {
+                        blittableJson = CreateNewOptionBlittableJsonReaderObject(blittableJson, jsonOperationContext, nameof(options.OperateOnTypes), oldOperateOnType);
+                    }
+
+                    var cmd = new ShardedStreamCommand(this, async stream =>
+                    {
+                        using (var gzipStream = new GZipStream(GetInputStream(stream, options), CompressionMode.Decompress))
+                        {
+                            var firstLoop = true;
+                            await gzipStream.CopyToAsync(exportOutputStream);
+                        }
+
+                    }, blittableJson);
+
+                    await ShardedContext.RequestExecutors[i].ExecuteAsync(cmd, jsonOperationContext);
+                }
+                await exportOutputStream.WriteAsync(Encoding.UTF8.GetBytes("}"));
+            }
+
+            return null;
```

Comment:
Missing the result

Reply:
will be handled in separate PR after marge

Reply:
done